gym/lib/gym/xcodebuild_fixes/generic_archive_fix.rb

```diff
@@ -0,0 +1,29 @@
+module Gym
+  class XcodebuildFixes
+    class << self
+      # Determine IPAs for the Watch App which aren't inside of a containing
+      # iOS App and removes them.
+      #
+      # In the future it may be nice to modify the plist file for the archive
+      # itself so that it points to the correct IPA as well.
+      #
+      # This is a workaround for this bug
+      # https://github.com/CocoaPods/CocoaPods/issues/4178
+      def generic_archive_fix
+        UI.verbose "Removing Orphaned WatchKit2 Applications"
```

Comment:
Perhaps change this to "Looking Orphaned WatchKit2 Applications" and then inside the if block below you can log "Removing Orphaned WatchKit2 Application #{app_path}".